### PR TITLE
feat: keyring access & Monereo address generation

### PR DESCRIPTION
**Pull request by brianp**
Description
---
Revamp the way we handle keyring access and use it to store the monero seed and the tari seed passphrase.

Motivation and Context
---
The user could deny access to the system Keychain, if they do we wouldn't be able to successfully store needed credentials for the app. Provide a wrapper to encapsulate storing the credentials in either the system keychain, or fallback to file IO storage. 

How Has This Been Tested?
---
Manually

What process can a PR reviewer use to test or verify this change?
---
Scenario 1:
- Install a slightly older version of TU. As of today (Friday 22nd, Nov)  this could be <=0.7.4.
- Assign a custom monero address / or leave the default monero address
- Then install the new version of the app
- Verify that your Monero address is your custom or the default monero address
- Verify no monero seed words are present

Scenario 2:
- Perform a completely fresh install of this version of TU
- Verify the monero address is already a custom address
- Verify you have access to the monero seed words

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

**Comment by brianp** [comment]
@mmrrnn @MCozhusheck please do a code review. As I'll be looking to merge this today, and get it out of my life :D

**Comment by mmrrnn** [comment]
Tested successfully on windows and linux
image

File: public/locales/en/settings.json [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "placeholder": "Enter Monero address",
     "title": "Monero address"
   },
+  "monero-seed-words": "Monero Seed Words",
```
Comment (mmrrnn) [inline public/locales/en/settings.json:64]: `npm run translate settings monero-seed-words="Monero Seed Words"` to generate for all languages

**Review by mmrrnn — APPROVED** [review]

**Review by MCozhusheck — APPROVED** [review]

**Comment by Tas4tari** [comment]
Tested successfully on Windows 11, app version 0.8.2 for:
- Did a fresh install of the app and saw a unique monero address and seed words
- I pasted in another monero address and confirmed that the seed words are no longer displayed after the app restarted

I will test the first scenario after completing the rest of the PR's

**Pull request merged**